packages/nx/src/tasks-runner/task-orchestrator.ts

```diff
@@ -802,13 +807,16 @@ export class TaskOrchestrator {
     this.runningTasksService.addRunningTask(task.id);
     this.runningContinuousTasks.set(task.id, childProcess);
 
-    childProcess.onExit(() => {
+    childProcess.onExit(async (code) => {
       if (this.tuiEnabled) {
         this.options.lifeCycle.setTaskStatus(task.id, NativeTaskStatus.Stopped);
       }
       if (this.runningContinuousTasks.delete(task.id)) {
         this.runningTasksService.removeRunningTask(task.id);
       }
+
+      const status = code === 0 ? 'success' : 'failure';
```

Comment:
An unexpected exit (`!this.cleaningUp`) should always be considered a failure. Continuous tasks are intended to run continuously until all dependent tasks have finished.